pkg/command/recommendation/options.go

```diff
@@ -170,6 +175,60 @@ func (opts *ContainerResourcesOptions) Apply(configuration *[]applications.Confi
 	}
 }
 
+// HPAResourcesOptions contains options for building the recommender configuration
+// for optimizing hpa resources.
+type HPAResourcesOptions struct {
+	TargetUtilizationMax map[string]int
+	TargetUtilizationMin map[string]int
+}
+
+func (opts *HPAResourcesOptions) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().StringToIntVar(&opts.TargetUtilizationMax, flagHPATargetUtilizationMax, opts.TargetUtilizationMax, "per-hpa resource max target utilization as `resource=value`; resource is one of: cpu")
+	cmd.Flags().StringToIntVar(&opts.TargetUtilizationMin, flagHPATargetUtilizationMin, opts.TargetUtilizationMin, "per-hpa resource min target utilization as `resource=value`; resource is one of: cpu")
+}
+
+func (opts *HPAResourcesOptions) Apply(configuration *[]applications.Configuration) {
+	lazyHPAResources := func() *applications.HPAResources {
+		if len(*configuration) == 0 {
+			*configuration = append(*configuration, applications.Configuration{HPAResources: &applications.HPAResources{}})
+		}
+		if (*configuration)[0].HPAResources == nil {
+			(*configuration)[0].HPAResources = &applications.HPAResources{}
+		}
+		return (*configuration)[0].HPAResources
```

Comment:
This breaks the `lazyContainerResources` (and the assumption that `configurations` only contains a single element): you are going to end up setting both container and HPA configurations on the same element instead of having individual elements in the configuration list; e.g. this would be valid:

```
configurations:
- containerResources:
    selector: x=y
- hpaResources:
    targetUtilitization:
      min:
        cpu: 92
```

But this would not:

```
configurations:
- containerResources:
    selector: x=y
  hpaResources:
    targetUtilitization:
      min:
        cpu: 92
```

Reply:
changed the `ContainerResourcesOptions` to `ConfigurationOptions` and added the HPA options within there.  I'm assuming that's the desired approach, if not let me know.